boa/src/builtins/error/mod.rs

```diff
@@ -93,8 +96,33 @@ impl Error {
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/toString
     #[allow(clippy::wrong_self_convention)]
     pub(crate) fn to_string(this: &Value, _: &[Value], context: &mut Context) -> Result<Value> {
-        let name = this.get_field("name").to_string(context)?;
-        let message = this.get_field("message").to_string(context)?;
-        Ok(format!("{}: {}", name, message).into())
+        if !this.is_object() {
+            return context.throw_type_error("'this' is not an Object");
+        }
+        let name = this.get_field("name");
+        let name_to_string;
+        let name = if name.is_undefined() {
+            "Error"
+        } else {
+            name_to_string = name.to_string(context)?;
+            name_to_string.as_str()
+        };
+
+        let message = this.get_field("message");
+        let message_to_string;
+        let message = if message.is_undefined() {
+            ""
+        } else {
+            message_to_string = message.to_string(context)?;
+            message_to_string.as_str()
+        };
+
+        if name == "" {
+            Ok(Value::from(message))
+        } else if message == "" {
+            Ok(Value::from(name))
+        } else {
+            Ok(Value::from(format!("{}: {}", name, message)))
+        }
```

Comment:
```suggestion
        if name.is_empty() {
            Ok(message.into())
        } else if message.is_empty() {
            Ok(name.into())
        } else {
            Ok(format!("{}: {}", name, message).into())
        }
```
better to check with `.is_empty()` than to compare ` == ""`, also it looks cleaner with `.into()`, instead of `from`